xblock/fields.py

```diff
@@ -465,7 +473,7 @@ def _calculate_unique_id(self, xblock):
         for the field in its given scope.
         """
         key = scope_key(self, xblock)
-        return hashlib.sha1(key).hexdigest()
+        return hashlib.sha1(key.encode('utf-8')).hexdigest()
```

Comment:
I believe what's happening is that python2 will autoconvert unicode to bytes (using the ascii codec), and this code relied on that.  Of course that would have broken on non-ascii characters, but scope_key seems to be pretty careful to restrict its output set.